### PR TITLE
Fix NDFT tests

### PR DESCRIPTION
**Pull request by chaithyagr**
The fixes include:
1) fftshift added for FFT <-> NDFT matrix test
2) Added a test to compare NDFT vs NUFFT. This is crucial and sets the tone for autodiff modules.
3) Adding normalize to support `ortho`, this is important as mostly all NUFFT is normalized. 
4) Removed a test `test_ndft_grid_matrix`, which tries o get the FFT matrix and match to a gridded NDFT matrix. But this is too complex and not generalizable currently. The test seems redundant so I removed it.

### CODE REVIEW

File: tests/test_ndft.py [hunk 1]
```diff
@@ -13,32 +13,9 @@
 
 from case_trajectories import CasesTrajectories, case_grid1D
 from helpers import assert_almost_allclose
+from mrinufft import get_operator
 
 
-@parametrize_with_cases(
```
Comment (chaithyagr) [inline tests/test_ndft.py:18]: This test is bad. Removed it :P

File: tests/test_ndft.py [hunk 2]
```diff
@@ -76,7 +53,32 @@ def test_ndft_implicit1(kspace, shape):
     linop_coef = implicit_type1_ndft(kspace, random_kspace.flatten(), shape)
     matrix_coef = matrix.conj().T @ random_kspace.flatten()
 
-    assert np.allclose(linop_coef, matrix_coef)
+    assert_almost_allclose(linop_coef, matrix_coef, atol=1e-4, rtol=1e-4, mismatch=5)
```
Comment (chaithyagr) [inline tests/test_ndft.py:56]: This test is good, so added it  :P

**Comment by paquiteau** [comment]
Good catch ! could you detail a bit which things were wrong (and why) ?
Also, lint your code ;)

**Comment by chaithyagr** [comment]
@paquiteau Ill let you add this test for CD! I think that its good practice to run it before every release. (More importantly, we need to make it run right now to make sure the tests run fine.)

**Comment by chaithyagr** [comment]
Can i merge when green? @paquiteau

**Pull request merged**